packages/material-ui-lab/src/Timeline/Timeline.js

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { capitalize } from '@material-ui/core/utils';
+import { withStyles } from '@material-ui/core/styles';
+import TimelineContext from './TimelineContext';
+
+export const styles = () => ({
+  /* Styles applied to the root element. */
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    padding: '6px 16px',
+    flexGrow: 1,
+  },
+  /* Styles applied to the root element if `align="left"`. */
+  alignLeft: {},
+  /* Styles applied to the root element if `align="right"`. */
+  alignRight: {},
+  /* Styles applied to the root element if `align="alternate"`. */
+  alignAlternate: {},
+});
+
+const Timeline = React.forwardRef(function Timeline(props, ref) {
+  const { align = 'left', classes, className, ...other } = props;
+
+  return (
+    <TimelineContext.Provider value={{ align }}>
+      <ul
+        className={clsx(classes.root, classes[`align${capitalize(align)}`], className)}
+        ref={ref}
+        {...other}
+      />
+    </TimelineContext.Provider>
+  );
+});
+
+Timeline.propTypes = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit the d.ts file and run "yarn proptypes"     |
+  // ----------------------------------------------------------------------
+  /**
+   * The position where the timeline should appear.
```

Comment:
This is a bit confusing for me. The timeline is being referred to as the visible line in the documentation so how can it have alternating alignment? I think this is what caused me some initial confusion when looking at the right aligned timeline example - I instantly associated alignment with text so when the text was aligned on the left it seemed weird until I read the description. Maybe something like eventAlignment= left | right | alternating would be better, I'm not sure.

cc @oliviertassinari

Reply:
I got the inspiration for this prop from other UI libraries actually...
- ant design has the property `mode="right"` which is displaying the same thing
- react suite also has the align property - `align="right"` for the similar thing

Maybe `eventAlignment` is more specific and won't introduce this confusion... Anyway, let me know you final thoughts and I can update it.

Reply:
@joshwooding From my perspective, we are good. If you consider how the CSS text-align property work, it sounds identical.

Reply:
> text-align specifies in the name what is being aligned

Interesting, I guess we might have abused on the shorthand with TableCell.align and Typography.align prop names.